qa-tests-backend/src/main/groovy/util/CollectorUtil.groovy

```diff
@@ -12,13 +19,68 @@ class CollectorUtil {
     static final String ENABLED_VALUE = "ENABLED"
     static final String DISABLED_VALUE = "DISABLED"
 
+    static parseJsonToProtobuf(String json) {
+        sensor.Collector.CollectorConfig.Builder builder = sensor.Collector.CollectorConfig.newBuilder()
+        JsonFormat.parser().merge(json, builder)
+        return builder.build()
+    }
+
+    static introspectionQuery(String collectorAddress, String endpoint) {
+        String uri = "http://${collectorAddress}${endpoint}"
+        HttpURLConnection connection = new URL(uri).openConnection()
+
+        // this might be unneeded?
+        connection.setRequestMethod("GET")
+
+        try {
+            connection.connect()
+
+            if (connection.responseCode != HttpURLConnection.HTTP_OK) {
+                throw new RuntimeException("IntrospectionQuery failed with ${connection.responseMessage}")
+            }
+            def jsonResponse = connection.inputStream.text
+            return parseJsonToProtobuf(jsonResponse)
+        } catch (Exception e) {
+            throw new RuntimeException("Error making request: ${e.message}", e)
+        } finally {
+            connection.disconnect()
+        }
+    }
+
+    static waitForConfigToHaveState(OrchestratorMain orchestrator, String state, int timeoutSeconds = 90, int port = 8080) {
+        def portForwards = orchestrator.createCollectorPortForwards(port)
+
+        log.info "Waiting for Collector Config propagation (${portForwards.size()} pods)"
+        int intervalSeconds = 1
+        int waitTime
+        def startTime = System.currentTimeMillis()
+        for (waitTime = 0; waitTime <= timeoutSeconds / intervalSeconds; waitTime++) {
+            if (portForwards.size() == 0) {
+                break
+            }
+
+            // if a pod has the right config, remove it from the list
+            // we need to check
+            portForwards.removeAll {
+                def config = introspectionQuery("127.0.0.1:${it.getLocalPort()}", "/state/runtime-config")
+                return config.networking.externalIps.enabled.name() == state
+            }
+            sleep intervalSeconds * 1000
+        }
+
+        // if we timed out, some collectors have not updated
+        // the config, so return false
+        return (waitTime <= timeoutSeconds / intervalSeconds)
+    }
 
-    static enableExternalIps(OrchestratorMain orchestrator) {
+    static enableExternalIps(OrchestratorMain orchestrator, int timeoutSeconds = 90) {
         setExternalIps(orchestrator, ENABLED_VALUE)
+        waitForConfigToHaveState(orchestrator, ENABLED_VALUE, timeoutSeconds)
```

Comment:
should we have a log of how long it took to enable?

Reply:
Done